XenModel/CustomFields/CustomFieldsManager.cs

```diff
@@ -155,25 +192,14 @@ public static string GetCustomFieldKey(CustomFieldDefinition customFieldDefiniti
             return CUSTOM_FIELD_BASE_KEY + CUSTOM_FIELD_DELIM + customFieldDefinition.Name;
         }
 
-        private static void RemoveCustomFieldsFrom(Session session, IEnumerable<IXenObject> os, CustomFieldDefinition customFieldDefinition)
-        {
-            InvokeHelper.AssertOffEventThread();
-
-            string customFieldKey = GetCustomFieldKey(customFieldDefinition);
-
-            foreach (IXenObject o in os)
-            {
-                Helpers.RemoveFromOtherConfig(session, o, customFieldKey);
-            }
-        }
-
         private static void SaveCustomFields(Session session, IXenConnection connection, List<CustomFieldDefinition> customFields)
         {
             Pool pool = Helpers.GetPoolOfOne(connection);
             if (pool != null)
             {
-                String customFieldXML = GetCustomFieldDefinitionXML(customFields);
-                Helpers.SetGuiConfig(session, pool, CUSTOM_FIELD_BASE_KEY, customFieldXML);
+                string customFieldXML = GetCustomFieldDefinitionXML(customFields);
+                Pool.remove_from_gui_config(session, pool.opaque_ref, CUSTOM_FIELD_BASE_KEY);
+                Pool.add_to_gui_config(session, pool.opaque_ref, CUSTOM_FIELD_BASE_KEY, customFieldXML);

```

Comment:
`set_gui_config` is faster because it involves one call. however, it sets the whole dictionary property, not just a value in it and it may have different permissions from add/remove (theoretically more restrictive since it changes multiple keys with potentially different permissions at once).

Reply:
I see, let's leave it then. Better safe than sorry. I'd add this as a comment because it might not be immediately clear (just trying to avoid bugs introduced through future refactoring commits)